### ISSUE TITLE
proper seeding

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/parallelMap#43]
Bernd wanted to open an issue as mentioned [here](https://github.com/mlr-org/mlr/issues/938). 
We might want to improve reproducibility in parallelMap by using seeding-mechanics for the different parallization techniques.

**Comment by pat-s** [comment]
Should we finally close this issue by incorporating the approach of https://stackoverflow.com/a/51347058/4185785?

I am afraid that many people messed up reproducibility in the past just because they were not aware of this. Also I think many people are not aware that `set.seed()` does not work for parallel calls.

Related issues

- https://github.com/mlr-org/parallelMap/issues/57
- https://github.com/mlr-org/mlr/issues/938

### Suggested solution

* Set `set.seed(1, "L'Ecuyer-CMRG")` before all calls of mode "multicore"

* Set `parallel::clusterSetRNGStream(iseed = 1)` after all calls of mode "socket" 

* What about other modes such as "mpi" and "batchtools"?

`parallelStart()` would also gain an arg to turn this behavior off. By default it would ensure reproducibility in parallel scenarios without requiring knowledge by the user about different RNG kinds.

@berndbischl @jakob-r 

---

On a side note: How does _future_ deal with that issue when using `plan(multicore)` or `plan(multisession)`? @mllg 
(better discuss this one in a separate issue in mlr3)
